geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -6392,6 +6401,23 @@ void basicDestroy(final EntryEventImpl event, final boolean cacheWrite, Object e
   }
 
   /**
+   * A set operation now can bootstrap a transaction now. User need to specifically disable this by
+   * setting this system property to true to get the old behavior.
+   *
+   * @Since Geode 1.3.0
+   */
+  public static final String PREVENT_SET_OP_BOOTSTRAP_TRANSACTION =
+      "preventSetOpBootstrapTransaction";
+
+  protected static boolean isSetOpBootstrapTransactionDisabled() {
+    return Boolean
+        .getBoolean(DistributionConfig.GEMFIRE_PREFIX + PREVENT_SET_OP_BOOTSTRAP_TRANSACTION)
```

Comment:
How about refactoring this into a method on DistributionConfig or some new SystemPropertyHelper class?
You could call it "boolean getProductBooleanProperty(String name)".
I would not say that if either is true you should return true.
Instead I think you should test to see if the geode one is exists. If it does use its value (true or false).
If it does not exist then check for the gemfire one.
To check for existence you can call "System.getProperty". It will return null if that property is not defined.

Reply:
SystemPropertyHelper class is used now.